### PR TITLE
Add tests for #1868

### LINKED ISSUE #1868: Atom index out of range error
Converting RXN blocks into reaction smiles via an rdkit reaction, the atom index is out of range. This seems to occur randomly, as multiple attempts on the same reaction block show that it may indeed be converted. 
The getIdx() method seems to be retrieving atomic indexes way beyond those that should be in the created molecule.  

https://github.com/rdkit/rdkit/blob/2190c991f03d44d0df1fd22b372ca85036074301/Code/GraphMol/ROMol.cpp#L158

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes #1868 

I believe this bug was fixed, but this explicitly tests for it.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@greglandrum Added a test for #1868, looks like it has been fixed.

File: Code/GraphMol/FileParsers/MolFileParser.cpp [hunk 1]
```diff
@@ -2479,6 +2479,75 @@ bool ParseV2000CTAB(std::istream *inStream, unsigned int &line, RWMol *mol,
   return fileComplete;
 }
 
+void finishMolProcessing(RWMol *res, bool chiralityPossible, bool sanitize,
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/MolFileParser.cpp]: Shouldn't this be named finishMDLMolProcessing?  Will it be used elsewhere?

Comment (bp-kelley) [reply Code/GraphMol/FileParsers/MolFileParser.cpp]: @greglandrum - a small comment here for a future thought.

Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolFileParser.cpp]: Sorry, I had done a review and then forgot to close it out.
This is only used in the mol file parser and will only be used there.
The naming isn't great, but it's consistent with the rest of the functions here.

**Pull request merged**